### PR TITLE
Enhance command: "insights info"

### PR DESCRIPTION
**Pull request by xiangce**
- https://github.com/RedHatInsights/insights-core/issues/1645
  https://github.com/RedHatInsights/insights-core/issues/1645#issuecomment-468166815
  https://github.com/RedHatInsights/insights-core/issues/1645#issuecomment-469552049
- Limit to only outputs Parsers/Combiners by default
   Added "-S" option to output related `Specs` in addition
- Check the parent dir when the file name is "*" os "*.xx" in `glob_file`

### CODE REVIEW

**Comment by xiangce** [comment]
Sample output:
~~~
 15:18:58  liuxc  ~/work/insights/insights-core    enhance_insights_info  
(.python3) ❱ insights-info uname
insights.parsers.uname.Uname
insights.combiners.redhat_release.RedHatRelease
insights.combiners.redhat_release.redhat_release
insights.combiners.ipv6.IPv6

 15:19:02  liuxc  ~/work/insights/insights-core    enhance_insights_info  
(.python3) ❱ insights-info -d insights.combiners.redhat_release.RedHatRelease
Help on class RedHatRelease in module insights.combiners.redhat_release:

class RedHatRelease(builtins.object)
 |  RedHatRelease(uname, rh_rel)
 |
 |  Combiner class to check uname and redhat-release for RHEL major/minor
 |  version.
 |  Prefer uname to redhat-release.
 |
 |  Attributes:
 |      major (int): The major RHEL version.
 |      minor (int): The minor RHEL version.
 |      rhel (str): The RHEL version, e.g. '7.2', '7.5-0.14'
 |      rhel6 (str): The RHEL version when it's RHEL6, otherwise None
 |      rhel7 (str): The RHEL version when it's RHEL7, otherwise None
 |      rhel8 (str): The RHEL version when it's RHEL8, otherwise None
 |
 |  Raises:
 |      ParseException: If the version can't be determined even though a Uname
 |          or RedhatRelease was provided.
 |
 |  Examples:
 |      >>> rh_rel.rhel
 |      '7.2'
 |      >>> rh_rel.major
 |      7
 |      >>> rh_rel.minor
 |      2
 |      >>> rh_rel.rhel6 is None
 |      True
 |      >>> rh_rel.rhel7
 |      '7.2'
 |      >>> rh_rel.rhel8 is None
 |      True
 |
 |  Methods defined here:
 |
 |  __init__(self, uname, rh_rel)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors defined here:
 |
 |  __dict__
...
~~~

File: insights/tools/query.py [hunk 1]
```diff
@@ -130,28 +131,51 @@ def get_datasources():
...
     if isinstance(d, sf.glob_file):
-        return any(re.match(path, glob2re(_simple_file(pat))) and not d.ignore_func(path) for pat in d.patterns)
+        for pat in d.patterns:
+            # Set the last part of the glob_file as the search target
+            pat_sp = pat.split('/')
+            search_tgt = pat_sp[-1]
+            if search_tgt.startswith("*") and len(pat_sp) >= 2:
+                # if the last part is "*" or "*XXX", set the parent dir as search target
+                # e.g: xinetd_conf: "/etc/xinet.d/*.conf"
+                #      logrotate_conf: "/etc/logrotate.d/*"
```
Comment (lonicerae) [inline insights/tools/query.py]: Can we give relevant parser information when look up the qemu vm configs in `/etc/libvirt/qemu/`?

```
qemu_xml = glob_file(r"/etc/libvirt/qemu/*.xml")
```

 For example, want to analyze a file `rhel7u2.xml`. Or we should use the `qemu` as a key word for searching?

```
$ insights-info qemu
insights.parsers.qemu_xml.QemuXML
insights.parsers.qemu_conf.QemuConf

$ insights-info rhel7u2.xml
==>NULL
```

Comment (xiangce) [reply insights/tools/query.py]: @lonicerae - Good catching. 
But in fact, this situation has been considered. And, the `insights-info qemu` is the current solution for this situation. 
Think of the `*.conf`, currently, its candidate specs are as follows:
~~~
$ grep "\*.conf" insights/specs/default.py
    etc_journald_conf_d = glob_file(r"etc/systemd/journald.conf.d/*.conf")
    httpd_conf = glob_file(["/etc/httpd/conf/httpd.conf", "/etc/httpd/conf.d/*.conf"])
    httpd_conf_scl_httpd24 = glob_file(["/opt/rh/httpd24/root/etc/httpd/conf/httpd.conf", "/opt/rh/httpd24/root/etc/httpd/conf.d/*.conf"])
    httpd_conf_scl_jbcs_httpd24 = glob_file(["/opt/rh/jbcs-httpd24/root/etc/httpd/conf/httpd.conf", "/opt/rh/jbcs-httpd24/root/etc/httpd/conf.d/*.conf"])
    krb5 = glob_file([r"etc/krb5.conf", r"etc/krb5.conf.d/*.conf"])
    limits_conf = glob_file(["/etc/security/limits.conf", "/etc/security/limits.d/*.conf"])
    modprobe = glob_file(["/etc/modprobe.conf", "/etc/modprobe.d/*.conf"])
    pluginconf_d = glob_file("/etc/yum/pluginconf.d/*.conf")
    sysctl_conf_initramfs = simple_command("/bin/lsinitrd /boot/initramfs-*kdump.img -f /etc/sysctl.conf /etc/sysctl.d/*.conf")
    tmpfilesd = glob_file(["/etc/tmpfiles.d/*.conf", "/usr/lib/tmpfiles.d/*.conf", "/run/tmpfiles.d/*.conf"])
    usr_journald_conf_d = glob_file(r"usr/lib/systemd/journald.conf.d/*.conf")  # note that etc_journald.conf.d also exists
    virt_who_conf = glob_file([r"etc/virt-who.conf", r"etc/virt-who.d/*.conf"])
~~~
It's not a good experience to output all the `unrelated` parsers when querying `$ insights-info abc.conf`.

Comment (koalakangaroo) [reply insights/tools/query.py]: @lonicerae @xiangce - there are a bunch of files with ".xml" suffix. 
~~~
./sos_commands/sar/sa16.xml
./etc/firewalld/zones/public.xml
./etc/libvirt/qemu/networks/autostart/default.xml
./etc/libvirt/qemu/networks/default.xml
~~~
Running "insights-info rhel7u2.xml" may trigger wrong parsers unless we consider to use full path. This only happens to glob-typed specs like:

~~~
etc_journald_conf_d = glob_file(r"etc/systemd/journald.conf.d/*.conf")
initscript = glob_file(r"etc/rc.d/init.d/*
krb5 = glob_file([r"etc/krb5.conf", r"etc/krb5.conf.d/*.conf"])
ovirt_engine_confd = glob_file("/etc/ovirt-engine/engine.conf.d/*")
yum_repos_d = glob_file("/etc/yum.repos.d/*")
...
~~~

TBH, I don't have any better method on my mind to better @xiangce's implementation. Let's have a quick discussion today.

**Pull request merged**